app/views/menus/show.html.haml

```diff
@@ -7,9 +7,20 @@
       %br/
       - if current_user == @menu.restaurant.user #this is throwing a nil error
         = link_to 'Edit Menu', edit_menu_path(@menu)
-      - if !@menu.dishes.nil?
-        Dishes:
-        %br/
-        - @menu.dishes.each do |dish|
-          = link_to dish.name, dish
-          %br/
+      - unless @menu.dishes.nil?
+        - @dishes_by_category.keys.each do |category|
+          %div{id: category.downcase}
+            %h1= category
+            - @dishes_by_category[category].each do |dish|
+              %p= dish.name
+              %p= dish.description
+              %p= dish.allergies
+              %p= dish.ingredients
+              %p= dish.calories
+              %p= dish.price
+
+              - if can? :create, ShoppingCart
+                = link_to 'Add to cart', dish_add_item_path(dish), method: :post, class: 'btn btn-success'
+              - if can? :manage, Dish
```

Comment:
@Blokkinen I'm going to let you handle all the other comments on this commit (or whoever else in your group you want to bring in).